compiler/rustc_borrowck/src/diagnostics/explain_borrow.rs

```diff
@@ -67,65 +74,146 @@ impl<'tcx> BorrowExplanation<'tcx> {
     ) {
         match *self {
             BorrowExplanation::UsedLater(later_use_kind, var_or_use_span, path_span) => {
-                let message = match later_use_kind {
-                    LaterUseKind::TraitCapture => "captured here by trait object",
-                    LaterUseKind::ClosureCapture => "captured here by closure",
-                    LaterUseKind::Call => "used by call",
-                    LaterUseKind::FakeLetRead => "stored here",
-                    LaterUseKind::Other => "used here",
-                };
                 // We can use `var_or_use_span` if either `path_span` is not present, or both spans are the same
                 if path_span.map(|path_span| path_span == var_or_use_span).unwrap_or(true) {
                     if borrow_span.map(|sp| !sp.overlaps(var_or_use_span)).unwrap_or(true) {
-                        err.span_label(
-                            var_or_use_span,
-                            format!("{}borrow later {}", borrow_desc, message),
-                        );
+                        match later_use_kind {
+                            LaterUseKind::TraitCapture => {
+                                err.subdiagnostic(BorrowUsedLater::TraitCapture {
+                                    borrow_desc,
```

Comment:
Does `borrow_desc` include English words? If it does then this message might not be translatable, likewise the others that use this.